### PR TITLE
Followup edits to PR 2216

### PR DESCRIPTION
**Pull request by bfallonf**
#2216

Labeled 3.3, but doing it anyway.

@deads2k Can I get an ack I didn't change any of the meaning in the bullet points? Thanks.

### CODE REVIEW

File: architecture/additional_concepts/authentication.adoc [hunk 1]
```diff
@@ -180,15 +180,15 @@ $ oc create -f <(echo '
...
-* `client_secret` can be any of the API tokens for that service account (`oc sa get-token serviceaccount.name`)
-* `redirect_uris` come from any annotation on the service account that has the prefix `serviceaccounts.openshift.io/oauth-redirecturi.`.
-  For example: `serviceaccounts.openshift.io/oauth-redirecturi.one`.
+A link:../../admin_guide/service_accounts.html[service account] can be used as a
+constrained form of OAuth client. Service accounts can only request a subset of
+scopes that allow access to some basic user information and power inside of the
+service account's own namespace. Specifically, `user:info`, `user:check-access`,
+and `role:<anyrole>:<serviceaccount.namespace only>[:!]`.
+
+* `client_id` can be used for `system:serviceaccount:<namespace>:<serviceaccount.name>`.
```
Comment (deads2k) [inline architecture/additional_concepts/authentication.adoc]: A `client_id` is an identifier that is used to identify the client to the oauth server.  In this case, the value of the identifier is `system:serviceaccount:<namespace>:<serviceaccount.name>`.

**Comment by bfallonf** [comment]
Thanks @deads2k . I made a small tweak. Let me know if that's off the mark.

In the meantime, @adellape @ahardin-rh @tnguyen-rh @tpoitras Peer review?

**Comment by deads2k** [comment]
> Thanks @deads2k . I made a small tweak.

Looks good.

File: architecture/additional_concepts/authentication.adoc [hunk 2]
```diff
@@ -180,15 +180,15 @@ $ oc create -f <(echo '
...
-access to some basic user information and power inside of the service account's own namespace.
-Specifically,  `user:info`, `user:check-access`, `role:<anyrole>:<serviceaccount.namespace only>[:!]`.
-
-* `client_id` is `system:serviceaccount:<namespace>:<serviceaccount.name>`
-* `client_secret` can be any of the API tokens for that service account (`oc sa get-token serviceaccount.name`)
-* `redirect_uris` come from any annotation on the service account that has the prefix `serviceaccounts.openshift.io/oauth-redirecturi.`.
-  For example: `serviceaccounts.openshift.io/oauth-redirecturi.one`.
+A link:../../admin_guide/service_accounts.html[service account] can be used as a
+constrained form of OAuth client. Service accounts can only request a subset of
+scopes that allow access to some basic user information and power inside of the
```
Comment (tpoitras) [inline architecture/additional_concepts/authentication.adoc]: could you further define "power"? Perhaps "role-based power"? (IIUC)

**Comment by tpoitras** [comment]
Brice, your style fixes look fine, but I feel like this could benefit from an example configuration?

**Comment by bfallonf** [comment]
@tpoitras Thanks, I updated to your suggestion.

As for an example, I'm not sure. @deads2k Do you think this needs a specific example? There's an example SCC in the link in the first sentence, but Tim brings up a good point that this is slightly different.

**Comment by bfallonf** [comment]
@tpoitras @deads2k Actually, I should have done this sooner, but I took as look at this topic as a whole. What's here in this PR is consistent with the topic above in that it ties into the example by substituting the values in the bullet list. I don't think an example is needed here.

@tpoitras You're the one who asked for it, you satisfied with this?

**Comment by tpoitras** [comment]
@bfallonf yeah cool, LGTM :camel:

**Comment by bfallonf** [comment]
Cool. Thanks @tpoitras . I'll merge this.

**Comment by bfallonf** [comment]
[rev_history]
|xref:../architecture/additional_concepts/authentication.html[Additional Concepts -> Authentication]
|Added section on using Security Context Constraints for OAuth.
%

**Pull request merged**